modules/parsers/parser-hdbdd/src/main/java/com/sap/xsk/parser/hdbdd/symbols/context/ContextSymbol.java

```diff
@@ -40,17 +40,24 @@ public void define(Symbol sym) {
 
   @Override
   public Symbol resolve(String name) {
-    Symbol symbol = symbols.get(name);
-    if (symbol != null) {
-      return symbol;
-    }
+    if(name.contains(".")){
+      String[] identifiers = name.split("\\.");
+      Symbol outerContext = findSymbol(identifiers[0]);
 
-    // if not here, check any enclosing com.sap.xsk.parser.hdbdd.symbols.scope
-    if (getEnclosingScope() != null) {
-      return getEnclosingScope().resolve(name);
-    }
+      return resolveByContextArray(identifiers, outerContext);
+    } else {
+      Symbol symbol = symbols.get(name);
+      if (symbol != null) {
+        return symbol;
+      }
+
+      // if not here, check any enclosing com.sap.xsk.parser.hdbdd.symbols.scope
+      if (getEnclosingScope() != null) {
+        return getEnclosingScope().resolve(name);
+      }
 
-    return null; // not found
+      return null; // not found
```

Comment:
Shouldn't we log somewhere?

Reply:
Not sure about logging, I think the cases in which we get null here handle it in some sort of way